getter.go

```diff
@@ -149,55 +244,68 @@ func (g *getter) worker() {
 	for c := range g.getCh {
 		g.retryGetChunk(c)
 	}
-
 }
 
-func (g *getter) retryGetChunk(c *chunk) {
+func (g *getter) retryGetChunk(c *Chunk) {
 	defer g.wg.Done()
 	var err error
-	c.b = <-g.sp.get
+	c.Bytes = <-g.sp.get
 	for i := 0; i < g.c.NTry; i++ {
 		time.Sleep(time.Duration(math.Exp2(float64(i))) * 100 * time.Millisecond) // exponential back-off
 		err = g.getChunk(c)
 		if err == nil {
 			return
 		}
-		logger.debugPrintf("error on attempt %d: retrying chunk: %v, error: %s", i, c.id, err)
+		logger.debugPrintf("error on attempt %d: retrying chunk: %v, error: %s", i, c.Id, err)
+		//If there is a re-used response in this chunk, it should be discarded since an error was returned
+		c.response = nil
 	}
 	g.err = err
+	c.Error = err
+	g.readCh <- c //expose error to the chunk handler
```

Comment:
Does this trigger something on the readCh to quit?  I am just wondering if this exposes a new state where we have orphaned go routines because it is exiting out here now where before it would not